### DISCUSSION TITLE
Translation bug in Inter App settings - problem with scaling

### DISCUSSION THREAD

**Discussion by user-hash** [NightscoutFoundation/xDrip#3725] [Bug reports]
The bug occurs in the stable version dfcbe-802024.09.17 - Samsung S23

In Inter-App settings - Broadcast, there is a toggle button (usually ON-OFF) that is translated to the local language, despite the xDrip set in English. As the local name is longer than in English it has problem with scaling.

**Comment by Navid200** [comment]
Can you post a screenshot to help clarify what you are describing.
Can you state that in screenshot what it is and what it should be?

**Comment by user-hash** [comment]
Sure -  as the xDrip language is set as default the sliding button description should be just ON/OFF.
![Screenshot_20241020_100409_xDrip+](https://github.com/user-attachments/assets/9a31b048-de66-4c21-8bab-9a5c9676911d)

**Reply by Navid200** [reply]
Which item in your screenshot are you talking about?

**Reply by user-hash** [reply]
> Which item in your screenshot are you talking about?
The only sliding button without English names:
![{3E55CEB4-9D0A-42FB-97B8-3CCA27781148}](https://github.com/user-attachments/assets/4e2b6a4b-3077-42b2-88df-ee7bb8e50301)

**Reply by Navid200** [reply]
My xDrip language is set to default and I don't see that.
I would like to see how I can recreate this problem.  But, I cannot.  
Do you have any idea, other than buying a Samsung, how I can recrate this problem?

**Reply by user-hash** [reply]
> My xDrip language is set to default and I don't see that. I would like to see how I can recreate this problem. But, I cannot. Do you have any idea, other than buying a Samsung, how I can recrate this problem?

You could try setting the phone language to Slovenian. That's the only thing I can think of. Do you use English as a default phone language?

**Reply by Navid200** [reply]
I tried Italian and the on/off switch is still on/off.
<br/>  

Which one of these two languages should I choose?   
  
![Screenshot_20241021-074130](https://github.com/user-attachments/assets/63d01a3e-f1c0-4543-b639-74d48b448d27)

**Reply by Navid200** [reply]
Are you using xDrip from a different repository?

**Reply by user-hash** [reply]
Please choose the second one. I am not sure about the repo, but I installed xDrip from the official source on GitHub. I had some problems with previous settings, so this is a new install.

**Reply by Navid200** [reply]
I have even tried this with Android 15 and cannot reproduce it.
Can you show a screenshot of your classic status page?

**Reply by user-hash** [reply]
> I have even tried this with Android 15 and cannot reproduce it. Can you show a screenshot of your classic status page?

![Screenshot_20241021_155428_xDrip+](https://github.com/user-attachments/assets/e9a4f5f9-6108-49ab-84bd-fb124de61478)

**Reply by Navid200** [reply]
Can you set the phone to automatically switch to landscape mode when you rotate it?
Then, rotate it.  What does that page look like in landscape mode?

**Reply by user-hash** [reply]
> Can you set the phone to automatically switch to landscape mode when you rotate it? Then, rotate it. What does that page look like in landscape mode?

I checked yesterday and the behavior was similar to the portrait mode. So label still in Slovene language, button is 2x width of other buttons, but no text scaling problem.

When I checked it today just to be sure, the button is now in English like it should be...
So looks like the bug is not appearing for me anymore.

**Reply by Navid200** [reply]
This is very strange.
Even if I wanted to change the code to translate the on-off switch, I wouldn't know how!
The switch is identical for all the items on that page.  That's why this was so strange to me.
And now that the problem is gone, it is strange still.

Let us know if it returns.

**Comment by user-hash** [comment]
Bug is back. I agree with you that it is strange, but luckily it is not game-breaking.

**Reply by Navid200** [reply]
I cannot prove it.  But, I see a lot of mess caused by Samsung.

**Reply by Navid200** [reply]
If you scroll up and down on that same page, is that the only switch with this problem?

**Reply by user-hash** [reply]
> If you scroll up and down on that same page, is that the only switch with this problem?

Hey, yes this is the only switch with the problem on this page. But I have just found exactly the same problem with the switch on "NFC Scan features"

**Comment by Navid200** [comment]
OK, I can recreate the problem.
Let me see what I can do about it.

**Comment by Navid200** [comment]
This should be fixed now if you use the release 2024.11.08 or any after.
Thanks for reporting this.
Please let us know if there are still any problems with it.

**Comment by user-hash** [comment]
Thank you for the quick solution, I will check the new version and get back.

**Comment by user-hash** [comment]
I have upgraded to the latest Nightly version from today -eb920d2 and it fixes the Broadcast switch.
The same problem is with the switch on "NFC Scan features" page.
What was the recipe to replicate this error? I am really intrigued.

![{50876FD5-435E-49B7-96E3-3CC1BD9F3F9F}](https://github.com/user-attachments/assets/57812aea-8b7c-4740-9e59-59fc06613dd0)

**Reply by Navid200** [reply]
We have a function in xDrip that communicates with translators to use a short form of yes or no.  
When you set xDrip to force English.  xDrip uses the translation keys and since you have chosen English, yes or no will be used.

There are a few cases where we have forgotten to use that function.  In that case, xDrip has no idea that it should translate.  So, it uses to local language.
We need to fix all the cases.

Would you please give me a list as far as you can of course so that I can fix them all in one shot?

**Comment by user-hash** [comment]
Here you go, I tried to get through all the menu settings, but I could still miss some.

Alarms, alerts: Calibration Alerts: First and last switch
Alarms, alerts: Extra Alerts: Persistant HIgh: Both Switches
Alarms, alerts: Extra Alerts: Forecasted low: 1st switch
Alarms, alerts: Extra Alerts: Other alerts: All 3 switches
Cloud upload: Nightscout sync: Both Switches
Cloud upload: MOngoDB: 1st switch
Cloud upload: influx: 1st switch

Android wear integration: 2nd 
BlueJay: 3rd and 4th 
Lefun: 3rd
MiBand: every switch

LEss common: Blue reader settings - all 3 switches
Libre 2: both 2 switches

**Reply by Navid200** [reply]
Thanks for this.
I figured out how to search for all of them and have a submission to fix them all:
https://github.com/NightscoutFoundation/xDrip/pull/3763

Thanks for your help.

**Comment by user-hash** [comment]
Wow, you are amazing. I did not think this bug would appear on other menus and you usually don't browse menu trees if everything is set up.
Did you manage to take a look at https://github.com/NightscoutFoundation/xDrip/discussions/3724 ?

 I think it is actually a bug if you look at the duplicated values on the graph and I don't think they were there when I was using the 640G uploader.
Does the duplicated entries happen only on Care Link Follower Collector?  As duplicated I mean that there is a separate bubble for carbs and insulin treatment, but values/information is only added to one. 
The graph could still be cleaned to show more info. Also, the orange for carbs and purple for IoB looks really nice and clearly visible - mockup below. 
![{1BC05C4B-D2EC-4561-8562-5C3A2B2E8A95}](https://github.com/user-attachments/assets/41df6db5-803b-4e1c-a4c8-36672bc86e48)

**Reply by Navid200** [reply]
I don't have any experience with carelink or with any simulator functions of xDrip.
